src/Tribe/Rewrite.php

```diff
@@ -412,18 +412,25 @@ protected function add_hooks() {
 	 */
 	public function filter_url_to_postid( $url ) {
 
+		$events_url = Tribe__Events__Main::instance()->getLink();
+
 		// check if the site is using pretty permalinks
 		if ( '' !== get_option( 'permalink_structure' ) ) {
 			$url_query = @parse_url( $url, PHP_URL_QUERY );
 
 			// Remove the "args" in case we receive any
 			if ( ! empty( $url_query ) ) {
 				$url = str_replace( '?' . $url_query, '', $url );
+			} else {
+				// Check if they're viewing the events page with pretty params
+				if ( substr( $url, 0, strlen( $events_url ) ) === $events_url ) {
```

Comment:
Could we just do `0 === strpos( $url, $events_url )`? (or `stripos` to be case-insensitive)

Reply:
Yes, we can do that. It's kind of the same.